docs/src/main/asciidoc/intro.adoc

```diff
@@ -6,10 +6,74 @@ Spring Cloud Sleuth implements a distributed tracing solution for http://cloud.s
+annotations (tags), the ID of the span that caused them, and process ID's (normally IP address).
 
-Spans are started and stopped, and they keep track of their timing information.  Once you create a span, you must stop it at some point in the future.
+Spans are started and stopped, and they keep track of their timing information.  Once you create a
+span, you must stop it at some point in the future.
 
-*Trace:* A set of spans forming a tree-like structure.  For example, if you are running a distributed big-data store, a trace might be formed by a put request.
+*Trace:* A set of spans forming a tree-like structure.  For example, if you are running a distributed
+big-data store, a trace might be formed by a put request.
 
+*Annotation:*  is used to record existence of an event in time. Some of the core annotations used to define
+the start and stop of a request are:
 
+    - *cs* - Client Sent - The client has made a request. This annotation depicts the start of the span.
+    - *sr* - Server Received -  The server side got the request and will start processing it.
+    If one subtracts the cs timestamp from this timestamp one will receive the network latency.
+    - *ss* - Server Sent -  Annotated upon completion of request processing (when the response
+    got sent back to the client). If one subtracts the sr timestamp from this timestamp one
+    will receive the time needed by the server side to process the request.
+    - *cr* - Client Received - Signifies the end of the span. The client has successfully received the
+    response from the server side. If one subtracts the cs timestamp from this timestamp one
+    will receive the whole time needed by the client to receive the response from the server.
+
+Visualization of what *Span* and *Trace* will look in a system together with the Zipkin annotations:
+
+image::trace-id.png[Trace Info propagation]
+
+Yellow note signifies a span. If you have such information in the note:
+
+[source]
+Trace Id = X
+Span Id = D
+Client Sent
+Client Received
+
+That means that the current span has *Trace-Id* set to *X*, *Span-Id* set to *D*. It also has emitted
+ *Client Sent* and *Client Received* events.
+
+=== Purpose
+
+In the following sections the example from the image above will be taken into consideration.
+
+==== Distributed tracing with Zipkin
+
+Altogether there are *10 spans* . If you go to traces in Zipkin you will see this number:
+
+image::zipkin-traces.png[Traces]
+
+However if you pick a particular trace then you will see *7 spans*:
+
+image::zipkin-ui.png[Traces Info propagation]
+
+NOTE: When picking a particular trace you will see merged spans. That means that if there were 2 spans sent to
+Zipkin with Server Received and Server Sent annotations then they will presented as a single span.
+
+You don't have to use Zipkin to profit from tracing. When grepping the logs of those four applications by trace id
```

Comment:
this comment should be moved to the next system, and I wouldn't necessarily suggest this is profitable :) rather that if you aren't running a zipkin system, you can still perform log correlation.